src/irc/core/irc-servers-setup.c

```diff
@@ -69,7 +69,14 @@ static void sig_server_setup_fill_chatnet(IRC_SERVER_CONNECT_REC *conn,
 		return;
 	g_return_if_fail(IS_IRCNET(ircnet));
 
-	if (ircnet->nick != NULL) g_free_and_null(conn->alternate_nick);
+	if (ircnet->nick != NULL) {
```

Comment:
This is already done in `server_setup_fill_chatnet`.

Reply:
Should I revert line 72 to what it was or remove it entirely? The check on `ircnet->nick` and subsequent free on `conn->alternate_nick` didn't make sense to me, and as you pointed out it's already done elsewhere.

Reply:
I don't have an answer to that question, sorry.
I think the logic behind the code goes like "if a chatnet specifies `nick` then try hard to use it instead of the global `alternate_nick`" and since the latter is now chatnet-specific you can probably just nuke it.